proposals/js-promise-integration/Overview.md

```diff
@@ -177,4 +261,34 @@ A function is suspendable if it was
+
+## Frequently Asked Questions
+
+1. **What is the purpose of the `Suspender` object?**
+
+   The `Suspender` object is used to connect a `Promise` returning import with a `Promise` returning export. Without this explicit connection, it becomes problematic especially when constructing so-called chains of modules: where one module calls into the exports of another.
+
+1. **Why do we try to prevent JavaScript programs from using this API?**
+
+   JavaScript already has a way of managing computations that can suspend. This is semantically connected to JavaScript `Promise` objects and the `async` function syntax. However, a more important reason is that it is important, in the context of JavaScript, that we do not introduce language features that can affect the behavior of existing programs.
+
+1. **Why does the API only apply t `WebAssembly.Function` values, and not `Function`s?**
+
+   The core interface elements in this API refer to the use of `WebAssembly.Function` entities rather than plain `Function` entities. There are some potential questions and issues about this choice:
+
+    * Compared to regular imports, it is arguably less ergonomic to manually mark the types of a function when importing a JavaScript function as a wrapped import. This is not needed for normal imports, but we are requiring it for wrapped imports.
+    * A wrapped import function is not callable from JavaScript. The role of a wrapped import is fundamentally to signal intentions to the process of instantiating modules. It is arguable that using wrapper functions in this way may not be the best architectural approach to addressing this signaling.
+
+1. **Which is better: the shared `Suspender` object or the static functions?**
+
+   Using a shared `Suspender` object that connects exports and imports at module instantiation time is significantly easier to use than the static version. The reason for this is that wrapping imports and exports using `WebAssembly.suspendOnReturnedPromise` and `WebAssembly.returnPromiseOnSuspend` require that the `Suspender` object is threaded through the computation. In particular, a `Suspender` object must be given as an additional argument to the export, and that that same `Suspender` object must be presented to the import. This requires some internal reorganization of the WebAssembly module to ensure the transmission of the object&mdash;which is an `externref` from the perspective of WebAssembly and so managing this may be complex.
+
+   However, the WebAssembly wrappers allow for program reentrancy&mdash;which makes it simpler to construct responsive applications. We anticipate that hand built WebAssembly applications will likely use the simpler `Suspender.suspendOnReturnedPromise` API and compilers that already know how to compile asynchronous code will use the `WebAssembly.suspendOnReturnedPromise` API.
```

Comment:
From this I assume that Emscripten might want to use the static API? However I'm not sure how it would help yet. In https://github.com/emscripten-core/emscripten/pull/16779 things seem to work ok with the suspender API? Apologies if I've missed some previous discussion on this, but it's the first I hear of the static API.

Reply:
I believe that Emscripten will use both variants of the API. For simple async/await, the suspender.* version will be sufficient. For the fibers API the static version will be needed (along with handling the shadow stack somehow).

Reply:
Interesting. I've been assuming that Fibers support in Emscripten would wait for full wasm stack switching anyhow. Wouldn't that be more efficient for pure wasm usage to avoid JS entirely?

Reply:
I see, sounds good.